### PR TITLE
Feat/movie results page

### PR DESCRIPTION
**Pull request by zkevkev**
## Description
This branch adds a movies_search service and movie PORO for use in the user discover results page. It implements these to make an API call and retrieve top rated movies and search movies by title from the TMDB API.
## Type of change
- [ ] fix
- [x] feat
- [x] test
- [ ] refactor
- [ ] docs
## Checklist
- [x] code has been self reviewed
- [x] code runs without any errors
- [x] thorough testing has been implemented if adding feature
- [x] all tests pass

### CODE REVIEW

File: spec/features/users/discover/results/index_spec.rb
Comment (zkevkev) [inline spec/features/users/discover/results/index_spec.rb:1]: Nice work on the testing for limiting results! Counting the average rating was a smart workaround. Really appreciate you help on this story.

File: app/views/users/discover/results/index.html.erb
Comment (zkevkev) [inline app/views/users/discover/results/index.html.erb:1]: Thanks for solving the limiting of results using `.take` you definitely taught me something here and I really appreciate it.

**Review by delaneymiranda1 — APPROVED** [review]
Good work on the movies search implementation. We'll refactor the link to the the welcome page into the application later!

**Review by zkevkev — COMMENTED** [review]
Great to collaborate with you on this story!

**Pull request merged**